### ISSUE TITLE
成為 Python AI 深度學習達人的第一堂課_note

### ISSUE THREAD

**Issue by pspiskinghuayuworld** [pspiskinghuayuworld/learn_note#4]
## 第一周
* 神經網路三個 **重要 model** ： NN、CNN、RNN
* 深度學習的核心技術：神經網路
* AI 創建過程：
   1.定義問題
   2.數學建模
   3.準備資料，分成訓練資料與測試資料
   4.架構神經網路並調整參數
   5.學習：運用loss function看和觀察值差多少
* 神經網路含：輸入層、隱藏層、輸出層
* 三層以上的隱藏層即屬於深度學習
* 神經元：接受若干個輸入，但僅會有一個輸出(即便輸出到不同神經元)
* 總刺激：輸入做加權和→ W1X1 + W2X2 + W3X3 +......
* 總刺激會加調整值：偏值(bias)
* 激活函數 activation functions 的幾種選擇：
   1. ReLU →最常用，現在神經網路的王牌，小於零就是零，大於零的時候，就是原來的數字。
   2. Sigmoid →模擬人類神經元的樣子，輸入的總刺激小，輸出就是零，收到的很大，總輸出就是1
   3. Gaussian →
* 當一個神經網路結構決定、activation functions 也決定，可以調就是權重(weights)和偏值(bias)
* 「最好」、「最近」就是 loss function 的值最小
* 因切線斜率指向(局部)最大值的方向，要將loss(L(w))調到最小，需要將權重(w)往(局部)極小值移動(切線斜率加上負號)
* 移動時避免一次移動太多，因此乘上很小的常數learning rate
* 看斜率的方向幫助神經網路學習之方法稱為 Gradient descent 梯度下降法
* 很多變數時偏微分
## 第二周
* SGD Stochastic Gradient Descent 將訓練順序打亂
* 模型名稱.compile( loss=你想要用什麼樣的損失函數, optimizer=你想要用什麼方式訓練你的模型 ,metrics=你想要增加什麼評分標準)
